platform/saivpp/vpplib/vppxlate/SaiVppXlate.c

```diff
@@ -647,6 +696,79 @@ static int config_lcp_hostif (vat_main_t *vam, vl_api_interface_index_t if_idx,
     return ret;
 }
 
+static void compute_mac_addr(u32 instance, u8 *mac_address)
+{
+    u64 addr = 0x000000000000;
+    u8 *mac = (u8 *)&addr;
+
+    mac[0] = 0x00;
+    mac[1] = 0x00;
+    mac[2] = 0x00;
+
+    for (int i = 5; i > 2; i--) {
+        u8 quotient = instance % 256;
+        mac[i] = quotient;
+        instance /= 256;
+    }
+
+    if (instance > 0) {
+        mac[2] += instance & 0xff;
+        mac[1] += (instance >> 8) & 0xff;
+    }
+
+    memcpy(mac_address, mac, 6);
+}
+
+static int __create_loopback_instance (vat_main_t *vam, u32 instance)
+{
+    vl_api_create_loopback_instance_t *mp;
+    int ret;
+    u8 mac_address[6] = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+    __plugin_msg_base = interface_msg_id_base;
+
+    M (CREATE_LOOPBACK, mp);
+    mp->client_index = htonl(socket_client_main.client_index);
```

Comment:
There is no need to explicitly set client_index here. The M macro already does that for you.

Reply:
Fixed